### PR TITLE
Sign up II - integration

### PR DESCRIPTION
**Pull request by Julianhm9612**
## Summary

Added new color variables
Added messages component
Added whitSpinner HOC to Button component
Added and configure toaster dependency
Added user service and i18n for generic messages
Integrated signup service

## Screenshots

signup success

signup success

signup error

## Trello Card

https://trello.com/c/U4sEIQ5U/23-sign-up-ii-integration

### CODE REVIEW

File: src/screens/SignUp/index.tsx [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
+import { useLazyRequest } from 'hooks/useRequest';
+import Messages from 'components/Messages';
+
+import { signUp } from '../../services/UserService';
 
 import styles from './styles.module.scss';
 
 function SignUp() {
   const { t, i18n } = useTranslation();
+  const [showError, setShowError] = useState(false);
```
Comment (ebedoyaalzate) [inline src/screens/SignUp/index.tsx]: You can use the error property from useRequest

File: src/services/UserService.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { ApiResponse } from 'apisauce';
+
+import api from 'config/api';
+import { User } from 'types/user';
+import { objectCamelToSnake } from 'utils/case';
+
+export async function signUp(user: User): Promise<ApiResponse<any>> {
+  const response = await api.post('/users', objectCamelToSnake(user));
```
Comment (ebedoyaalzate) [inline src/services/UserService.ts]: You don't need to use async/await just return the promise

File: src/services/UserService.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { ApiResponse } from 'apisauce';
+
+import api from 'config/api';
+import { User } from 'types/user';
+import { objectCamelToSnake } from 'utils/case';
+
+export async function signUp(user: User): Promise<ApiResponse<any>> {
```
Comment (ebedoyaalzate) [inline src/services/UserService.ts]: Don't use any

**Review by ebedoyaalzate — CHANGES_REQUESTED** [review]

File: src/services/UserService.ts (hunk 2 above)
Comment (Julianhm9612) [reply src/services/UserService.ts]: Has been changed :godmode:

File: src/screens/SignUp/index.tsx (hunk 1 above)
Comment (Julianhm9612) [reply src/screens/SignUp/index.tsx]: Has been changed :godmode:

File: src/services/UserService.ts (hunk 3 above)
Comment (Julianhm9612) [reply src/services/UserService.ts]: Has been enhanced with interfaces :suspect:

**Review by ebedoyaalzate — APPROVED** [review]

**Pull request merged**